modules/nw-ovn-k-anp-best-practices.adoc

```diff
@@ -0,0 +1,36 @@
+/ Module included in the following assemblies:
+//
+// * list of assemblies where this module is included:
+// networking/network_security/AdminNetworkPolicy/nw-ovn-k-anp-recommended-practices.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="anp-best-practices_{context}"]
+= Designing AdminNetworkPolicy
+
+When building `AdminNetworkPolicy` (ANP), you should consider the following when creating your policies:
+
+//This will need to be added in at 4.16.1
+
+//* You should create `AdminNetworkPolicy` at different priorities so that precedence is well defined. It is not guaranteed which policy will take precedence when overlapping ANP are created.
+
+* There is only one `BaselineAdminNetworkPolicy` for the cluster so be careful in defining it and you might consider using it as a default guardrail policy.
```

Comment:
I think "there is only one" and "be careful" are separate pieces of advice here:

1. There is only one BANP, and especially with the current version of the API which doesn't have "tenancy", there are only a handful useful things you can do with it, and it would probably help if we explained them:
   - You can set a "default deny" policy for ingress in user namespaces, so that users have to add NetworkPolicies to allow the ingress traffic that they want to allow, and if they don't do that it will be denied.
   - You can set a "default deny" policy for egress in user namespaces, so that users have to add NetworkPolicies to allow the egress traffic that they want to allow, and if they don't do that it will be denied.
     - The above, but also with an "allow" rule to allow egress to DNS by default so users don't have to remember to do that themselves.
   - You can set an egress policy that allows egress to all pods and denies access to `0.0.0.0/0` and `::/0` in user namespaces, so that users still have allow-egress-by-default for cluster-internal traffic, but have deny-by-default for cluster egress traffic, and they have to add NetworkPolicies to allow the cluster egress traffic that they want to allow.
If you don't want to do one of those things, then you _probably_ don't want to use BANP.

2. "Be careful" that you scope your BANP so that it only denies traffic in _user_ namespaces, not _system_ namespaces, because the system namespaces aren't going to have NPs to override your BANP. (Um... I'm not sure exactly how you do this? You need to select on the existence of some label which is part of the default project template...)

Reply:
Ok, I tried to work this out in some rewording of a new section. Dan, thank you for review.